wip/toolchains-1.0.buildspec

```diff
@@ -0,0 +1,27 @@
+# Central Repository coordinates for the Reference release (for multi-module, pick an arbitrary module)
+groupId=nl.basjes.experiment
+artifactId=toolchains
+version=1.0
+
+display=${groupId}:${artifactId}
+
+# Source code
+gitRepo=https://github.com/nielsbasjes/ToolChainsInCiBuilds.git
+gitTag=main
+
+# Rebuild environment prerequisites
+tool=mvn-3.8.7
+# All LTS JDK versions and JDK-17 is the default (used to starts Maven with)
+jdk=lts
```

Comment:
I'm not happy with this part yet.
Indicating `lts` makes it unclear which exact versions are needed. I expect in a few years that the term `lts` means something different (i.e. no more jdk 8).
So the exact set of needed JDK versions must be made explicit.

Also because of the timezone and umask issues I think these also must be made explicit. 
If unspecified I want to set 
- the timezone to `Australia/Eucla`: a real inhabited timezone with (as far as I can tell) the smallest population (~200 people).
- the umask to 0055 : Which makes absolutely no sense in a real situation.

So either a project clearly indicates the needed values for reproduction, or the project has all plugins configured correctly and it builds in a reproducible way. 

In general: If it is relevant for the reproducibility it must be specified in the buildspec config.

So I'm now thinking about something like:
```
# The jdk under which maven must be run
jdk=17
# The jdk versions available via toolchains
toolchain=8|11|17
# The default timezone that is to be used by the JDK (i.e. java -Duser.timezone=..)
timezone=Europe/Amsterdam
# The umask that must be set during the build
umask=0002
```

Reply:
that's why in the second step, I propose to name your specific image, and there will be other images

on umask, it's another topic: it will be another step

we need to go step by step